src/shogun/regression/GaussianProcessRegression.cpp

```diff
@@ -37,37 +37,9 @@ std::shared_ptr<RegressionLabels> GaussianProcessRegression::apply_regression(st
 	require(m_method->supports_regression(), "{} with {} doesn't support "
 			"regression",	m_method->get_name(), lik->get_name());
 
-
-	std::shared_ptr<RegressionLabels> result;
-
-	// if regression data equals to NULL, then apply regression on training
-	// features
-	if (!data)
-	{
-		std::shared_ptr<Features> feat;
-
-		// use inducing features for FITC inference method
-		if (m_method->get_inference_type()==INF_FITC_REGRESSION)
-		{
-			auto fitc_method = m_method->as<FITCInferenceMethod>();
-			feat=fitc_method->get_inducing_features();
```

Comment:
ah  nevermind. This basically fell back to predicting on the inducing features in case there was no training data provided. But that doesnt make sense anyways...it should just predict on the data provided and if a user wants to predict at the inducing points only, he has to pass those into the train method.